app/controllers/posts_controller.rb

```diff
@@ -1,11 +1,14 @@
 class PostsController < ApplicationController
   def index
-    @user = User.find(params[:user_id])
+    # @user = User.includes(:posts).where(user: {id: params[:user_id]}))
+    @user = User.includes(:posts).find(params[:user_id])
```

Comment:
- [ ] In this `posts#index` action, It is where we **fetch all the posts by a particular user together with comments** This is the typical case for the n+1 problem. I appreciate your efforts in trying to refactor your codes in order to solve it, however, It looks like you are not fetching posts for a particular user together with respective comments. Kindly, implement the correct logic. 😎 😃 👍 
💡 HINT: It should look like the following: 
```ruby
@posts = @user.posts.includes(:comments)
```